unit tests use tas pipeline

miyunari opened the pull request:
This PR should run the Cosign unit tests with the Konflux pipeline. 
I used these PRs as examples and guidelines:
https://github.com/securesign/trillian/pull/197
https://github.com/securesign/fulcio/pull/175
https://github.com/securesign/gitsign/pull/145

jira: https://issues.redhat.com/browse/SECURESIGN-1295

JasonPowr commented:
You need to update the cosign on-push and pull to use the new task, eg: https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L366-L375

JasonPowr commented on .tekton/client-server-cg-pull-request.yaml:
```diff
@@ -8,6 +8,7 @@ metadata:
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
     pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch == "main" && ( "Dockerfile.client-server-cg.rh".pathChanged() || ".tekton/client-server-cg-pull-request.yaml".pathChanged() || "trigger-konflux-builds.txt".pathChanged() )
+    pipelinesascode.tekton.dev/task: "[.tekton/cosign-unit-test.yaml]"
```
This should only be needed in `.tekton/cosign-pull-request.yaml` and `.tekton/cosign-pull-request.yaml`

miyunari replied:
Oh ok thank you, I removed it in the other ones :)

miyunari commented:
> You need to update the cosign on-push and pull to use the new task, eg: https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L366-L375

Ok ty, I added it - and hopefully in the correct way - could you check again @JasonPowr ? :)

miyunari commented on .tekton/client-server-cg-pull-request.yaml:
```diff
@@ -176,14 +162,20 @@ spec:
-        workspace: workspace
       - name: basic-auth
         workspace: git-auth
     - name: prefetch-dependencies
```
Thanks! I think it was only one - I rechecked every file :) Sry, I lost the overview :sweat_smile:

JasonPowr commented on .tekton/client-server-cg-pull-request.yaml:
```diff
@@ -340,6 +327,8 @@ spec:
         value: $(tasks.build-container.results.IMAGE_DIGEST)
       - name: image-url
         value: $(tasks.build-container.results.IMAGE_URL)
+      - name: SOURCE_ARTIFACT
```
This Param is in the wron place it should be part of the `sast-snyk-check-oci-ta` task eg https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L298

Best to check the other one's too :)

miyunari commented:
@JasonPowr I hope I addressed all of the points you suggested in the right way? I am also super unsure if I chose the right quay images... :/ Do you know why the tests are still failing or what else could be wrong? Thanks for your help already :hugs:

JasonPowr commented on .tekton/client-server-f-pull-request.yaml:
```diff
@@ -181,33 +163,32 @@ spec:
+          value: prefetch-dependencies-oci-ta
         - name: bundle
-          value: quay.io/konflux-ci/tekton-catalog/task-prefetch-dependencies:0.1@sha256:058a59f72997c9cf1be20978eb6a145d8d4d436c6098f2460bd96766bb363b20
+          value: quay.io/konflux-ci/tekton-catalog/task-prefetch-dependencies-oci-ta:0.1@sha256:57979e1c289bfe09acb70401f35558a9032e749b398a43fea049c044f9d96afe
```
```suggestion
          value: quay.io/redhat-appstudio-tekton-catalog/task-prefetch-dependencies-oci-ta:0.1@sha256:57979e1c289bfe09acb70401f35558a9032e749b398a43fea049c044f9d96afe
```

see: https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L186

JasonPowr commented:
@miyunari The review comments I just made should help to resolve the pipelines not running :),

miyunari commented:
> @miyunari The review comments I just made should help to resolve the pipelines not running :),

Thanks a lot for all of your help! It seems the pipeline needs a lot of time to run, is that normal? And two of the Konflux steps already failed - https://github.com/securesign/cosign/pull/244/checks?check_run_id=29302067611

JasonPowr commented on .tekton/client-server-cg-push.yaml:
```diff
@@ -376,23 +356,21 @@ spec:
         operator: in
         values:
         - "false"
+    - name: run-unit-test
```
We dont need to run the unit tests in these pipelines we only need to run them in `.tekton/cosign-pull-request.yaml` and `.tekton/cosign-push.yaml`

miyunari replied:
Oh sry,  misunderstood that part :) I will change it :+1:

JasonPowr commented:
@miyunari Sometimes when there is a lot of load on the konflux servers it can act strange i.e taking ages for a pipeline to run, The ones that have run I am seeing a failure in the unit tests for cosign eg https://console.redhat.com/application-pipeline/workspaces/rhtas/applications/cli/pipelineruns/client-server-cg-on-pull-request-fqcws


And no worries :) happy to help

JasonPowr commented:
/retest

JasonPowr commented on .tekton/client-server-re-pull-request.yaml:
```diff
@@ -176,33 +180,31 @@ spec:
-      workspaces:
-      - name: source
-        workspace: workspace
     - name: build-container
```
Missing some changes here eg https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L215

Please check the others :)

JasonPowr commented on .tekton/client-server-pull-request.yaml:
```diff
@@ -166,14 +144,18 @@ spec:
+      - name: ociArtifactExpiresAfter
+        value: $(params.image-expires-after)
       runAfter:
       - init
```
You need to remove the output workspace for the clone-repository task see: https://github.com/securesign/gitsign/blob/main/.tekton/gitsign-pull-request.yaml#L138

Please check the others :)

miyunari replied:
Hmm I am not really sure what you mean with this point? I looked at your example and compared it with the marked lines and can't see what needs to be changed :)

miyunari replied:
Thanks - removed it now :)

JasonPowr commented on .tekton/cosign-unit-test.yaml:
```diff
@@ -5,16 +5,46 @@ metadata:
+    - name: cachi2
+      emptyDir: {}
 
 # This file bundles the unit tests for cosign. 
```
You can remove these comments
